### ISSUE TITLE
🐛 Bug Report: Auth provider for Google fails with "Invalid URL"

### ISSUE THREAD

**Issue by magnusp** [backstage/backstage#15242]
### 📜 Description

Using the Google authentication provider fails with "Invalid URL" when using the recent change for a "listen" directive.

### 👍 Expected behavior

Authenticating with Google should work

### 👎 Actual Behavior with Screenshots

![image](https://user-images.githubusercontent.com/1431685/207636516-25499bfe-63b2-48c3-9177-258fe9145d35.png)

### 👟 Reproduction steps

* Configure Google authentication according to the docs.
* Use the following configuration directives:
```
backend:
  baseUrl: http://localhost:7007
  listen: ':7007'
```

Authentication errors with "Invalid URL". I believe that this is different from the error "Invalid redirect_url".
![image](https://user-images.githubusercontent.com/1431685/207636970-49a66f1b-ca5b-4c24-a537-0f3215cf15ab.png)

### 📃 Provide the context for the Bug.

Using the older configuration directives works:
```
backend:
  baseUrl: http://localhost:7007
  listen:
    port: 7007
```
Other users have reported similar behaviour regarding the error: https://discord.com/channels/687207715902193673/978977570852835379/1047930052383354980

### 🖥️ Your Environment

Backstage 1.8.3 running locally

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by jhaals** [comment]
Sorry to hear that you're having problems.

Do you specify a callbackUrl in your auth section or not?
Do you have any additional information if you try authenticating with the browser dev tools open. Perhaps there's a failing request with some additional information.

**Comment by magnusp** [comment]
Hmm, no callbackUrl under my auth section. I can't find that as a documented property? I will follow up with results following https://backstage.io/docs/auth/troubleshooting#stepping-through-authentication-manually

**Comment by Anonyfox** [comment]
having the same issue with github auth as of today! it works on dev, but fails when deployed on prod with a real domain

**Comment by Rugvip** [comment]
This is likely due to a bug with the `:7007` listen address format. It's been fixed but the fix won't go out into main line until 1.9, which is tomorrow. A workaround in the meantime would be to for example configure it as `0.0.0.0:7007`.
